src/composed/signed_key/secret.rs

```diff
@@ -315,3 +315,23 @@ impl PublicKeyTrait for SignedSecretSubKey {
         self.key.to_writer_old(writer)
     }
 }
+
+impl From<&SignedSecretKey> for SignedPublicKey {
```

Comment:
I would prefer doing this on the non ref version and making the clone explicit on the caller side